### DISCUSSION TITLE
Mobile hotspot to use for internet with Tesla android auto?

### DISCUSSION THREAD

**Discussion by WillE726** [tesla-android/issue-tracker#240] [General]
Can we use our phones Wi-Fi hotspot to supple internet to the android?

**Comment by mikegapinski** [comment]
No and it will never be possible. It's a hardware limitation. The Wi-Fi interface is used by the hotspot with Tesla specific routing changes. Bluetooth tethering sucks so hard I would not recommend it anyone. This leaves us with USB tethering or a LTE modem with simcard

And no, I cannot add another wifi card via USB. Not something supported in Android to begin with

**Reply by WillE726** [reply]
Ok thanks.  For the LTE modem can you use a Wi-Fi hotspot if it has an Ethernet connection?

**Reply by stankuhn** [reply]
@mikegapinski I remember that in previous version (unfortunately can’t remember which exactly) i could share internet to android over bluetooth from my iphone. When I upgraded to latest version it is not possible anymore. When I pair it (which is now quite difficult actually manytimes pairing fails) there is no option to share internet in bluetooth device options. Is it somehow specifically disabled now? it didn’t seem to me so bad. and it is really much more comfortable than to plug phone to USB. I always hate it in other cars. carplay is wireless. but to get internet we need cable. that sucks 😉

**Reply by mikegapinski** [reply]
> @mikegapinski I remember that in previous version (unfortunately can’t remember which exactly) i could share internet to android over bluetooth from my iphone. When I upgraded to latest version it is not possible anymore. When I pair it (which is now quite difficult actually manytimes pairing fails) there is no option to share internet in bluetooth device options. Is it somehow specifically disabled now? it didn’t seem to me so bad. and it is really much more comfortable that to plug phone to USB. I always hate it in other cars. carplay is wireless. but to get internet we need cable. that sucks 😉

BT tether, ad2p and BLE was not working in Android 13 builds; next build will get it back to where it was on Android 12. Btw, BT tethering is so terribly slow that I cannot imagine using it. Most users migrate to LTE modems after using Tesla Android for a longer while

**Reply by stankuhn** [reply]
hmm. paying phone data. paying tesla premium connectivity. I’ve got usb modem but it would be 3rd data subscription 😕 I’ll stick with cable for a while. Any ETA for next build? is there a chance the BT tether will improve in future?

**Reply by mikegapinski** [reply]
> hmm. paying phone data. paying tesla premium connectivity. I’ve got usb modem but it would be 3rd data subscription 😕 I’ll stick with cable for a while. Any ETA for next build? is there a chance the BT tether will improve in future?

There is 0 sense in paying for premium connectivity if you use Tesla Android most of the time; you are not using it anyways. There is nothing to improve in terms of BT tethering, it's slow by design

**Reply by stankuhn** [reply]
remote cameras are good motivator for premium connectivity. also there are slight issues with android auto being on whenever the car is on. it will download big updates whenever it wants. also takes time until it boots up and everything starts working. I’ve got it on switch and turn it on when I want it. 
BT tether may get redesigned 😉but that’s obviously not in out hands. what about the next build any estimates? thanks.

**Reply by mikegapinski** [reply]
I left that part out of intentionally ;) It's generally not a good practice to ask for ETAs on free and open source software
